R/measures.R

```diff
@@ -63,28 +69,67 @@
 #' )
 #' }
 riem_measures <- function(
-    station = "VOHY",
-    date_start = "2014-01-01",
-    date_end = as.character(Sys.Date())) {
+    station,
+    date_start,
+    ...,
+    date_end = as.character(Sys.Date()),
+    data = "all",
+    elev = FALSE,
+    latlon = TRUE,
+    report_type = NULL) {
+  if (!rlang::is_character(station, n = 1)) {
+    cli::cli_abort("{.arg station} must be a string.")
+  }
   date_start <- format_and_check_date(date_start, "date_start")
+  rlang::check_dots_empty()
   date_end <- format_and_check_date(date_end, "date_end")
   if (date_end < date_start) {
     cli::cli_abort("{.arg date_end} must be bigger than {.arg date_start}.")
   }
+  if (!rlang::is_character(data, n = 1)) {
+    cli::cli_abort("{.arg data} must be a string.")
+  }
+  if (!is.logical(elev)) {
+    cli::cli_abort("{.arg elev} must be a logical (TRUE/FALSE)") # nolint: nonportable_path_linter
+  }
+  if (!is.logical(latlon)) {
+    cli::cli_abort("{.arg latlon} must be a logical (TRUE/FALSE)") # nolint: nonportable_path_linter
+  }
+
+  report_type <- report_type %||% c("routine", "specials")
+  report_type <- tolower(report_type) # not case-sensitive
+  report_type <- rlang::arg_match(
```

Comment:
I like your use of a map to abstract away the literals!

One side effect of `rlang::arg_match(multiple = TRUE)`, however, is that the error message can be misleading in terms of the number required:

```
! `report_type` must be one of "hfmetar", "routine", or "specials", not "3,4".
```

It's the `one of` that throws me for a mild loop.  I wonder if we could reduce any misleading by making line 12 a more clear comment, perhaps:

```
#' @param report_type A vector of strings, representing report types to query. The available options are "hfmetar", "routine", "specials". Default value is c("routine", "specials"). # nolint: line_length_linter
```